drivers/stm32_eth/stm32_eth.c

```diff
@@ -31,6 +31,14 @@ static mutex_t _tx = MUTEX_INIT;
 static mutex_t _rx = MUTEX_INIT;
 netdev_t *_netdev;
 
+#if IS_USED(MODULE_NETDEV_LINK_UP)
+#include "xtimer.h"
+#define TIMEOUT 1000000
+xtimer_t timer;
+static bool link_status = false;
+static bool first_call = true;
```

Comment:
Some style nit and `timer` should be static as well.

```suggestion
#include "xtimer.h"

#define TIMEOUT 1000000

static xtimer_t timer;
static bool link_status = false;
static bool first_call = true;
```

Also might safe some RAM by defining `link_status` and `first_call` as flags in an `uint8_t` instead of using full word width for storing binary values.

Reply:
This is not how I meant it, but ok. Using flags vs. two `uint8_t` most likely gives the latter a bit more advantage ROM-wise :sweat_smile:

Reply:
That's not what I meant, but ok. The way you did it is probably as good as flags RAM-wise, but a bit better ROM-wise than using flags.